From 791d26f58d02f1dca04965e1b46a90703237ea16 Mon Sep 17 00:00:00 2001
From: Alexey Puzhevich <apuzhevi@habana.ai>
Date: Thu, 08 May 2025 12:31:46 +0000
Subject: [PATCH] Revert "[FSW-13187] fs1: mem_copy: skip barrier for sim"

This reverts commit f65c14c80c779fbd9f5dd3e18c1e883c4efe663d.

Coral fix provided (FSW-14348) hence the workaround
no longer needed.

Change-Id: I1501081c8c39cdd7b0f6bf53ea955c3ddba23d9a
---

diff --git a/include/zephyr/drivers/misc/common/fs1/hl_platform_defs.h b/include/zephyr/drivers/misc/common/fs1/hl_platform_defs.h
index 2ce9ff9..2e43338 100644
--- a/include/zephyr/drivers/misc/common/fs1/hl_platform_defs.h
+++ b/include/zephyr/drivers/misc/common/fs1/hl_platform_defs.h
@@ -176,19 +176,16 @@
 	u32 __val, __addr;									\
 	int err;										\
 												\
-	/* Temporarily comment-out barrier for coral till issue SW-222241 is resolved */	\
-	if (!hl_is_sim_mode()) {								\
-		__addr = mmIO_0_CFG_HALF_0_PSOC_ARC_MNG_LOCAL_BASE +				\
-			mmARC_LOCAL_CONF_ARC_LBU_AXI_SPLIT_CTRL2;				\
-		err = readl_poll_busy_wait_timeout(__addr,					\
-				__val,								\
-				__val & ARC_LOCAL_CONF_ARC_LBU_AXI_SPLIT_CTRL2_NO_WR_INFLIGHT_M,\
-				HL_W_POLL_DELAY_US,						\
-				HL_W_BARRIER_TIMEOUT_US);					\
-		if (err)									\
-			LOG_ERR("HL_MNG_W_BARRIER timeout");					\
-	}											\
-done:												\
+	__addr = mmIO_0_CFG_HALF_0_PSOC_ARC_MNG_LOCAL_BASE +					\
+		 mmARC_LOCAL_CONF_ARC_LBU_AXI_SPLIT_CTRL2;					\
+	err = readl_poll_busy_wait_timeout(__addr,						\
+			__val,									\
+			__val & ARC_LOCAL_CONF_ARC_LBU_AXI_SPLIT_CTRL2_NO_WR_INFLIGHT_M,	\
+			HL_W_POLL_DELAY_US,							\
+			HL_W_BARRIER_TIMEOUT_US);						\
+	if (err)										\
+		LOG_ERR("HL_MNG_W_BARRIER timeout");						\
+												\
 	err;											\
 })
 
